CODIGO/TileEngine_Map.bas

```diff
@@ -17,26 +17,25 @@ Attribute VB_Name = "TileEngine_Map"
 '
 Option Explicit
 
-Sub SwitchMap(ByVal map As Integer)
+Sub SwitchMap(ByVal map As Integer, ByVal NewResourceMap As Integer)
```

Comment:
the whole point of instancing is to separate the resource of the map from the logical map, so we don't need to duplicate the resource every time we want to use the same map multiple times

now we have 100 identical maps when we could be using only one, this will become more important in future implementations

Reply:
I don't thing we should change `SwitchMap` in this way and add an extra argument. It's too complicated.

`SwitchMap` should do only one thing. Here the logic now is too complicated as it's not clear which map is loaded.

We should think a way to create utility functions to `CloneMap` and other things and use them before calling switch map.

Reply:
Or something like `LoadResourcesForExistingMap(byval map as integer)`

Reply:
we don't clone maps here

I don't think is complicated at all
the position has 2 components
1)the logical number that is referenced in the server
2) the resource file

until now both were the same (this forced us to store the map files as numbers) 
now we are saying "hey you are in the map 755 on the server that is using the resource file "200" do your setup"
this makes the client load the resource file 200 and setup the logical position on map 755

we cannot use a single param because we don't know the parity of map and resource until the server notify us